### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**

### CODE REVIEW

File: src/convertToObject.js [hunk 1]
```diff
@@ -3,16 +3,35 @@
...
+
+    if (key.trim().length !== 0) {
+      return {
+        ...newObject,
+        [key.toString().trim()]: [value].toString().trim(),
+      };
+    }
+
+    return { ...newObject };
+  };
```
Comment (ruslanpashkov) [inline src/convertToObject.js:30]: do we need to create a separate callback variable, when we can pass it directly in our `reduce`?

File: src/convertToObject.js [hunk 2]
```diff
@@ -3,16 +3,35 @@
...
-  // write your code here
+  const eachString = sourceString.split(';');
+
+  const callback = (newObject, stringStyle) => {
+    const [key, value] = stringStyle.split(':');
+
+    if (key.trim().length !== 0) {
+      return {
+        ...newObject,
+        [key.toString().trim()]: [value].toString().trim(),
```
Comment (ruslanpashkov) [inline src/convertToObject.js:25]: do we need to convert our properties to the string, when `typeof sourceString === 'string'`?

File: src/convertToObject.js [hunk 3]
```diff
@@ -3,16 +3,35 @@
...
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const eachString = sourceString.split(';');
+
+  const callback = (newObject, stringStyle) => {
+    const [key, value] = stringStyle.split(':');
+
+    if (key.trim().length !== 0) {
```
Comment (ruslanpashkov) [inline src/convertToObject.js:22]: we can use one of these options here

```suggestion
    if (key) {
```

```suggestion
    if (value) {
```

File: src/convertToObject.js (hunk 2 above)
Comment (ruslanpashkov) [inline src/convertToObject.js:25]: you don't need an array literal for value here

File: src/convertToObject.js [hunk 4]
```diff
@@ -3,16 +3,35 @@
...
+      return {
+        ...newObject,
+        [key.toString().trim()]: [value].toString().trim(),
+      };
+    }
+
+    return { ...newObject };
+  };
+
+  const result = eachString.reduce(callback, {});
```
Comment (ruslanpashkov) [inline src/convertToObject.js:32]: let's give a clearer name, to specify what data will be stored in our variable

**Review by ruslanpashkov — CHANGES_REQUESTED** [review]

**Review by naziamoff — APPROVED** [review]
🔥

**Pull request closed without merging**